airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -151,14 +151,16 @@ def ti_run(
                 "previous_state": previous_state,
             },
         )
-    log.info("Task with %s state started on %s ", previous_state, ti_run_payload.hostname)
+    else:
+        log.info("Task with %s state started on %s ", previous_state, ti_run_payload.hostname)
     # Ensure there is no end date set.
     query = query.values(
         end_date=None,
         hostname=ti_run_payload.hostname,
         unixname=ti_run_payload.unixname,
         pid=ti_run_payload.pid,
-        state=State.RUNNING,
+        state=TaskInstanceState.RUNNING,
```

Comment:
Fixed up a nit where we were using the generic `State` enum rather than the TI specific one. (no change in behaviour, this is just "more correct")